### PR TITLE
Fix single metric query for Prometheus format

### LINKED ISSUE #11364: Metrics are exposed in the format that is not compliant with Prometheus spec
Getting below exception while parsing the metrics that is available in path **/metrics** . I'm using prometheus-to-cloudwatch agent to scrap the metrics from my open liberty based microprofile application and to push cloudwatch.   During this process i'm getting below exceptions 

 **second TYPE line for metric name "base_gc_total", or TYPE reported after samples.** .  It looks like issue is due to duplication of the metric names which is mentioned below

base_gc_total
base_gc_time_seconds

Attached the complete metrics that is available in /metric path of the application
[CompleteAvailableMetrics.txt](https://github.com/OpenLiberty/open-liberty/files/4349039/CompleteAvailableMetrics.txt)

### PR DESCRIPTION
**Pull request by Channyboy**
Fix #11364 
#build

This was introduced from : https://github.com/OpenLiberty/open-liberty/pull/11592

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/writer/PrometheusMetricWriter.java [hunk 1]
```diff
@@ -123,7 +123,10 @@ protected void writeMetricMapAsPrometheus(StringBuilder builder, String registry
             double conversionFactor = conversionAppendEntry.getValue();
             String appendUnit = conversionAppendEntry.getKey();
 
-            if (metricMetadata.getTypeRaw().equals(MetricType.COUNTER)) {
+            //If current metadata that we are parsing does not have a matching metric... skip
+            if (currentMetricMap.isEmpty()) {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/writer/PrometheusMetricWriter.java:127]: Should the empty check on `crrentMetricMap` move to line 114.  If it is empty, you don't need to compute `description`, `unit`, `conversionFactory` etc.

File: dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java [hunk 2]
```diff
@@ -66,7 +66,10 @@ protected void writeMetricMapAsPrometheus(StringBuilder builder, String registry
             double conversionFactor = conversionAppendEntry.getValue();
             String appendUnit = conversionAppendEntry.getKey();
 
-            if (metricMetadata.getTypeRaw().equals(MetricType.COUNTER)) {
+            //If current metadata that we are parsing does not have a matching metric... skip
+            if (currentMetricMap.isEmpty()) {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java:70]: Should the empty check on crrentMetricMap move to line 56. If it is empty, you don't need to compute description, unit, conversionFactory etc.

**Comment by Channyboy** [comment]
#build 
#run-libby-bot

File: dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java [hunk 3]
```diff
@@ -66,10 +70,7 @@ protected void writeMetricMapAsPrometheus(StringBuilder builder, String registry
             double conversionFactor = conversionAppendEntry.getValue();
             String appendUnit = conversionAppendEntry.getKey();
 
-            //If current metadata that we are parsing does not have a matching metric... skip
-            if (currentMetricMap.isEmpty()) {
-                continue;
-            } else if (metricMetadata.getTypeRaw().equals(MetricType.COUNTER)) {
+            //if (metricMetadata.getTypeRaw().equals(MetricType.COUNTER)) {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java:73]: Uncomment the if statement

Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java]: ah right, may have pushed that commit a little bit late in the day :D

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Pull request merged**